sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/basicLogicalOperators.scala

```diff
@@ -182,14 +182,16 @@ case class Intersect(left: LogicalPlan, right: LogicalPlan) extends SetOperation
   }
 }
 
-case class Except(left: LogicalPlan, right: LogicalPlan) extends SetOperation(left, right) {
-
+abstract class ExceptBase(left: LogicalPlan, right: LogicalPlan) extends SetOperation(left, right) {
   /** We don't use right.output because those rows get excluded from the set. */
   override def output: Seq[Attribute] = left.output
 
   override protected def validConstraints: Set[Expression] = leftConstraints
 }
 
+case class Except(left: LogicalPlan, right: LogicalPlan) extends ExceptBase(left, right)
+case class ExceptAll(left: LogicalPlan, right: LogicalPlan) extends ExceptBase(left, right)
```

Comment:
We need a logical node for `ExceptAll`? As another option, we can add a flag as a field value of `Except` (`case class Except(left: LogicalPlan, right: LogicalPlan, all: Boolean)`? That's because `Except` and `ExceptAll` is not different for the analyzer.

Reply:
@maropu Right. So this way , most of the pattern matching happens on the the Base class where things are common. I went back and forth on this as well.. If there is a consensus i will change it.

Reply:
ok cc: @gatorsmile @cloud-fan

Reply:
Let us avoid adding a new logical plan node. : )